back/spec/finders/events_finder_spec.rb

```diff
@@ -90,4 +90,66 @@
       expect(result_record_ids).to match_array expected_record_ids
     end
   end
+
+  describe '#ongoing_during_condition' do
+    let_it_be(:event1) do
+      create(
+        :event,
+        start_at: Time.zone.parse('2023-01-01T00:00:00Z'),
+        end_at: Time.zone.parse('2023-01-02T00:00:00Z')
+      )
+    end
+
+    let_it_be(:event2) do
+      create(
+        :event,
+        start_at: Time.zone.parse('2023-01-01T00:00:00Z'),
+        end_at: Time.zone.parse('2023-01-03T00:00:00Z')
+      )
+    end
+
+    let_it_be(:event3) do
+      create(
+        :event,
+        start_at: Time.zone.parse('2023-01-10T12:00:00Z'),
+        end_at: Time.zone.parse('2023-01-10T14:00:00Z')
+      )
+    end
+
+    let_it_be(:events) { Event.where(id: [event1.id, event2.id, event3.id]) }
+
+    where(:start_at, :end_at, :expected_events) do
+      [
+        ['2022-12-31T12:00:00Z', '2023-01-01T00:00:00Z', []],
+        ['2022-12-31T12:00:00Z', '2023-01-01T01:00:00Z', [ref(:event1), ref(:event2)]],
+        ['2023-01-01T12:00:00Z', '2023-01-01T14:00:00Z', [ref(:event1), ref(:event2)]],
+        ['2023-01-01T12:00:00Z', '2023-01-04T00:00:00Z', [ref(:event1), ref(:event2)]],
+        ['2023-01-02T00:00:00Z', '2023-01-03T00:00:00Z', [ref(:event2)]],
+
+        ['2023-01-10T00:00:00Z', '2023-01-10T12:00:00Z', []],
+        ['2023-01-10T00:00:00Z', '2023-01-10T12:00:01Z', [ref(:event3)]],
+        ['2023-01-10T00:00:00Z', '2023-01-10T11:00:01-01:00', [ref(:event3)]],
+
+        ['2023-01-10T14:00:00Z', '2023-01-10T23:59:59Z', []],
+        ['2023-01-10T13:59:59Z', '2023-01-10T23:59:59Z', [ref(:event3)]],
+        ['2023-01-10T15:59:59+02:00', '2023-01-10T23:59:59Z', [ref(:event3)]]
+      ]
```

Comment:
Maybe add an range with a start date after the end date, for completeness? (I'd expect it to return `[]`)